snooty/parser.py

```diff
@@ -965,6 +982,70 @@ def validate_tabs_children(self, node: n.Directive) -> None:
         node.children = new_children
         return
 
+    def validate_io_code_block_children(self, node: n.Directive) -> None:
+        # new_children should contain input and output directives
+        new_children: List[n.Node] = []
+        line = node.start[0]
+        expected_children = {"input", "output"}
+
+        for child in node.children:
+            if isinstance(child, n.Directive):
```

Comment:
You can get rid of this level of indentation just by having `if not isinstance(child, n.Directive):`, handling that case, and `continue`ing